app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/general.ui.test.js

```diff
@@ -0,0 +1,347 @@
+const selectedRecipients = ['code'];
+
+const itemIdInSession = 'solution-1';
+const itemNameInSession = 'solution-name';
+const selectedRecipientIdInSession = 'recipient-1';
+const selectedRecipientNameInSession = 'recipient-name';
+const selectedPriceIdInSession = 'price-1';
+const catalogueSolutionIdInSession = 'solution-1';
+const deliveryDateInSession = '2020-10-10';
+const recipientsInSession = JSON.stringify(recipients);
+const selectedRecipientsInSession = JSON.stringify(selectedRecipients);
+
+const orderItemPageDataInSession = JSON.stringify({
+  itemId: itemIdInSession,
+  itemName: itemNameInSession,
+  serviceRecipientId: selectedRecipientIdInSession,
+  serviceRecipientName: selectedRecipientNameInSession,
+  selectedPrice,
+  recipients,
+  deliveryDate: deliveryDateInSession,
+  selectedRecipients,
+});
+
+const mocks = (mockSelectedPrice) => {
+  nock(solutionsApiUrl)
+    .get('/api/v1/prices/price-1')
+    .reply(200, mockSelectedPrice);
+};
+
+const defaultPageSetup = {
+  withAuth: true, getRoute: true, postRoute: false, mockData: selectedPrice,
+};
+const pageSetup = async (setup = defaultPageSetup) => {
+  if (setup.withAuth) {
+    await setState(ClientFunction)('fakeToken', authTokenInSession);
+  }
+  if (setup.getRoute) {
+    mocks(setup.mockData);
+    await setState(ClientFunction)('fakeToken', authTokenInSession);
+    await setState(ClientFunction)(sessionKeys.selectedRecipientId, selectedRecipientIdInSession);
+    await setState(ClientFunction)(sessionKeys.selectedRecipientName, selectedRecipientNameInSession);
+    await setState(ClientFunction)(sessionKeys.selectedItemId, itemIdInSession);
+    await setState(ClientFunction)(sessionKeys.selectedItemName, itemNameInSession);
+    await setState(ClientFunction)(sessionKeys.selectedPriceId, selectedPriceIdInSession);
+    await setState(ClientFunction)(sessionKeys.catalogueSolutionId, catalogueSolutionIdInSession);
+    await setState(ClientFunction)(sessionKeys.plannedDeliveryDate, deliveryDateInSession);
+    await setState(ClientFunction)(sessionKeys.recipients, recipientsInSession);
+    await setState(ClientFunction)(sessionKeys.selectedRecipients, selectedRecipientsInSession);
+  }
+  if (setup.postRoute) {
+    await setState(ClientFunction)(sessionKeys.orderItemPageData, orderItemPageDataInSession);
+  }
+};
+
+fixture('Catalogue-solutions - flat ondemand - general')
+  .page('http://localhost:1234/order/some-fake-page')
+  .afterEach(async (t) => {
+    await nockAndErrorCheck(nock, t);
+  });
+
+test('should render the solution table headings', async (t) => {
+  await pageSetup();
+  await t.navigateTo(pageUrl);
+
+  const solutionTable = Selector('div[data-test-id="solution-table"]');
+  const solutionNameColumnHeading = solutionTable.find('[data-test-id="column-heading-0"]');
+  const practiceSizeColumnHeading = solutionTable.find('[data-test-id="column-heading-1"]');
```

Comment:
Practice size is specific to per patient pricing so we should change this to `quantityColumnHeading`.

Reply:
Done